common/persistence/cassandra/metadata_store.go

```diff
@@ -161,22 +168,35 @@ func (m *MetadataStore) CreateNamespaceInV2Table(
 
 	previous := make(map[string]interface{})
 	applied, iter, err := m.session.MapExecuteBatchCAS(batch, previous)
+	deleteOrphanNamespace := func() {
+		// Delete namespace from `namespaces_by_id`
+		if errDelete := m.session.Query(templateDeleteNamespaceQuery, request.ID).WithContext(ctx).Exec(); errDelete != nil {
+			m.logger.Warn("Unable to delete orphan namespace record. Error", tag.Error(errDelete))
+		}
+	}
+
 	if err != nil {
```

Comment:
don't need to delete orphan row from `namespace_by_id` table, since we cannot be sure if it's an orphan row or the corresponding entry exists in `namepsaces`. This should be cleaned up later if `CreateNamespace` is called with the same `namespace` but a different `Id` if it's an orphan entry.